node/cli/src/service.rs

```diff
@@ -115,22 +115,24 @@ impl Default for NetworkConfig {
     }
 }
 
-pub fn start_service<S>(
+pub struct DevNetConfig {
+    /// how often devnet produces blocks, in milliseconds
+    pub block_period: Duration
+}
+
+impl Default for DevNetConfig {
+    fn default() -> Self {
+        DevNetConfig {
+            block_period: Duration::from_millis(100)
+        }
+    }
+}
+
+pub fn start_service(
     network_cfg: NetworkConfig,
     client_cfg: ClientConfig,
-    spawn_consensus_task_fn: S
-) where
-    S: Fn(
-            Receiver<Gossip<ChainPayload>>,
-            Receiver<ChainPayload>,
-            Sender<Gossip<ChainPayload>>,
-            Receiver<Control<BeaconWitnessSelector>>,
-            Sender<ChainConsensusBlockBody>,
-        ) -> ()
-        + Send
-        + Sync
-        + 'static,
-{
+    devnet_cfg: Option<DevNetConfig>,
```

Comment:
this seems weird. it may be better to have a spawn_consensus wrapper that takes the values needed from the devnet config and returns `S`.. thoughts @nearmax ?

Reply:
The advantage is that we can remove unnecessary arguments in `passthrough::spawn_consensus`.

Reply:
doesn't really seem like too much of an advantage to me.. i'm not sold either way though.  injection seems more natural here, but we can always refactor if this is an issue later

Reply:
My point is that in the future, `passthrough::spawn_consensus` might take arguments that are only used for devnet. It seems to me that trying to wrap `passthrough::spawn_consensus` and the real consensus spawner is not the best idea.

Reply:
k.. fine with this for now. something cleaner may be more obvious once @nearmax has done refactors like remove network from devnet.

Reply:
Yes, I think the architecture will be altered soon anyway.